mock/service/service.go

```diff
@@ -26,6 +31,46 @@ var Manifest = map[string]string{
 	"url": "https://github.com/kubernetes-csi/csi-test/mock",
 }
 
+// JavaScript hooks to be run to perform various tests
+type Hooks struct {
+	Globals                        string `yaml:"Globals"` // will be executed once before all other scripts
+	CreateVolumeStart              string `yaml:"CreateVolumeStart"`
+	CreateVolumeEnd                string `yaml:"CreateVolumeEnd"`
+	DeleteVolumeStart              string `yaml:"DeleteVolumeStart"`
+	DeleteVolumeEnd                string `yaml:"DeleteVolumeEnd"`
+	ControllerPublishVolumeStart   string `yaml:"ControllerPublishVolumeStart"`
+	ControllerPublishVolumeEnd     string `yaml:"ControllerPublishVolumeEnd"`
+	ControllerUnpublishVolumeStart string `yaml:"ControllerUnpublishVolumeStart"`
+	ControllerUnpublishVolumeEnd   string `yaml:"ControllerUnpublishVolumeEnd"`
+	ValidateVolumeCapabilities     string `yaml:"ValidateVolumeCapabilities"`
+	ListVolumesStart               string `yaml:"ListVolumesStart"`
+	ListVolumesEnd                 string `yaml:"ListVolumesEnd"`
+	GetCapacity                    string `yaml:"GetCapacity"`
+	ControllerGetCapabilitiesStart string `yaml:"ControllerGetCapabilitiesStart"`
+	ControllerGetCapabilitiesEnd   string `yaml:"ControllerGetCapabilitiesEnd"`
+	CreateSnapshotStart            string `yaml:"CreateSnapshotStart"`
+	CreateSnapshotEnd              string `yaml:"CreateSnapshotEnd"`
+	DeleteSnapshotStart            string `yaml:"DeleteSnapshotStart"`
+	DeleteSnapshotEnd              string `yaml:"DeleteSnapshotEnd"`
+	ListSnapshots                  string `yaml:"ListSnapshots"`
+	ControllerExpandVolumeStart    string `yaml:"ControllerExpandVolumeStart"`
+	ControllerExpandVolumeEnd      string `yaml:"ControllerExpandVolumeEnd"`
+	NodeStageVolumeStart           string `yaml:"NodeStageVolumeStart"`
+	NodeStageVolumeEnd             string `yaml:"NodeStageVolumeEnd"`
+	NodeUnstageVolumeStart         string `yaml:"NodeUnstageVolumeStart"`
+	NodeUnstageVolumeEnd           string `yaml:"NodeUnstageVolumeEnd"`
+	NodePublishVolumeStart         string `yaml:"NodePublishVolumeStart"`
+	NodePublishVolumeEnd           string `yaml:"NodePublishVolumeEnd"`
+	NodeUnpublishVolumeStart       string `yaml:"NodeUnpublishVolumeStart"`
+	NodeUnpublishVolumeEnd         string `yaml:"NodeUnpublishVolumeEnd"`
+	NodeExpandVolumeStart          string `yaml:"NodeExpandVolumeStart"`
+	NodeExpandVolumeEnd            string `yaml:"NodeExpandVolumeEnd"`
+	NodeGetCapabilities            string `yaml:"NodeGetCapabilities"`
+	NodeGetInfo                    string `yaml:"NodeGetInfo"`
+	NodeGetVolumeStatsStart        string `yaml:"NodeGetVolumeStatsStart"`
+	NodeGetVolumeStatsEnd          string `yaml:"NodeGetVolumeStatsEnd"`
```

Comment:
Traditionally, json and yaml identifiers start with lowercase (`globals`, `nodeGetVolumeStatsEnd`, ...)

Reply:
I meant for these to resemble the names of the methods... But... OK, I'll change it.